storybook/qmlTests/tests/tst_SwapInputPanel.qml

```diff
@@ -0,0 +1,193 @@
+            walletAssetStore: d.walletAssetStore
+        }
+    }
+
+    Component {
+        id: componentUnderTest
+        SwapInputPanel {
+            anchors.centerIn: parent
+            store: d.store
+        }
+    }
+
+    property SwapInputPanel controlUnderTest: null
+
+    TestCase {
+        name: "SwapInputPanel"
+        when: windowShown
+
+        function test_basicSetupAndDefaults() {
+            controlUnderTest = createTemporaryObject(componentUnderTest, root)
+            verify(!!controlUnderTest)
+            verify(controlUnderTest.width > 0)
+            verify(controlUnderTest.height > 0)
+
+            tryCompare(controlUnderTest, "swapSide", SwapInputPanel.SwapSide.Pay)
+            tryCompare(controlUnderTest, "caption", qsTr("Pay"))
+            tryCompare(controlUnderTest, "selectedHoldingId", "")
+            tryCompare(controlUnderTest, "cryptoValue", 0)
+            tryCompare(controlUnderTest, "cryptoValueRaw", "0")
+        }
+
+        function test_basicSetupReceiveSide() {
+            controlUnderTest = createTemporaryObject(componentUnderTest, root, {swapSide: SwapInputPanel.SwapSide.Receive})
+
+            verify(!!controlUnderTest)
+            verify(controlUnderTest.width > 0)
+            verify(controlUnderTest.height > 0)
+
+            tryCompare(controlUnderTest, "swapSide", SwapInputPanel.SwapSide.Receive)
+            tryCompare(controlUnderTest, "caption", qsTr("Receive"))
+            tryCompare(controlUnderTest, "selectedHoldingId", "")
+            tryCompare(controlUnderTest, "cryptoValue", 0)
+            tryCompare(controlUnderTest, "cryptoValueRaw", "0")
+        }
+
+        function test_basicSetupWithInitialProperties() {
+            controlUnderTest = createTemporaryObject(componentUnderTest, root,
+                                                     {
+                                                         swapSide: SwapInputPanel.SwapSide.Pay,
+                                                         inputTokenKey: "STT",
+                                                         inputTokenAmount: "10000000.0000001"
+                                                     })
+            verify(!!controlUnderTest)
+            waitForRendering(controlUnderTest)
+
+            tryCompare(controlUnderTest, "swapSide", SwapInputPanel.SwapSide.Pay)
+            tryCompare(controlUnderTest, "selectedHoldingId", "STT")
+            tryCompare(controlUnderTest, "cryptoValue", 10000000.0000001)
+            verify(controlUnderTest.cryptoValueValid)
+        }
+
+        function test_setTokenKeyAndAmounts_data() {
+            return [
+                { tag: "1.42", tokenAmount: "1.42", valid: true },
+                { tag: "0.00001", tokenAmount: "0.00001", valid: true },
+                { tag: "1234567890", tokenAmount: "1234567890", valid: true },
+                { tag: "1234567890.1234567890", tokenAmount: "1234567890.1234567890", valid: true },
```

Comment:
can we add test for "," separated input form user in some way?

Reply:
Yeah, we can change the locale on the fly to test with one that has `,` as a decimal separator, good idea! I'll add it

Reply:
Added the test